sample/samplers.go

```diff
@@ -85,13 +85,13 @@ func (s *Sampler) sample(tokens []token) (token, error) {
 	}
 
 	// topK also sorts the tokens in descending order of logits
-	tokens = topK(tokens, s.topK)
+	topK(&tokens, s.topK)
```

Comment:
Only topK needs to pass in the slice itself, and this is since it changes the size of the slice